src/messenger/messenger.cpp

```diff
@@ -516,9 +519,8 @@ void Messenger::ServerLoop() {
         bool has_custom_serverloop = poll_items.server_callbacks_[i] != nullptr;
         MESSENGER_LOG_TRACE("[PID={}] Messenger RECV-FR {} (custom serverloop: {}): {}", ::getpid(), msg.GetRoutingId(),
                             has_custom_serverloop, msg.GetRawPayload());
-        if (!has_custom_serverloop) {
-          ProcessMessage(msg);
-        } else {
+        ProcessMessage(msg);
```

Comment:
This change is needed if the user of Messenger wants to : 
1. Have its own router 
2. Register callback with the Messenger's Loop 

The issue for the current Messenger API is:
If I want to register a callback when `SendMsg`, I need to register it with the Messenger's main loop. But because I have a customized serverloop, I also need to evaluate and invoke the callback. 

I changed this to fa temp fix. I think there could be better ways to solve this, such as allowing the server's callback to get the `recv_cb_id` so that it could operate on the callback id.

Reply:
I thought we were going to roll it into #1331 but I don't see my changes there any more.

I noticed this while prototyping replication and changed it so that the callback function has direct access to the entire `ZmqMessage`, see [here](https://github.com/lmwnshn/noisepage/blob/replication/src/include/messenger/messenger_defs.h). Roughly speaking, every different component gets its own serverloop and implements its own callbacks rather than piggybacking on the main messenger loop.

I think the replication framework PR has been languishing for a few weeks. I can open a separate PR to bring in my changes, I think this would work for you?